distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso

```diff
@@ -148,9 +158,17 @@ make_join_kind_selector display=Display.Always =
 
 ## PRIVATE
    Make a join condition selector.
-make_join_condition_selector : Table -> Display -> Widget
-make_join_condition_selector table display=Display.Always =
-    col_names_selector = make_column_name_selector table display=Display.Always
+make_join_condition_selector : Table -> Display -> Function -> Widget
+make_join_condition_selector table display=Display.Always cache=Nothing =
+    right = if cache.is_nothing.not then
```

Comment:
Under what circumstances can the `cache` be Nothing?

Reply:
`cache` is `Nothing` when the function is called by some other code than `Widgets.get_widget_json` function. For example existing tests don't provide `cache`.

Reply:
If it's **only** for tests, shouldn't we start to require `cache` to be not Nothing and simply amend our tests? That seems like a better approach than maintaining a 'legacy' mode just for testing (especially as tests should probably test the new behaviour with cache anyway).

Reply:
Defaulted `cached` arguments makes the PR smaller. Some tests may and should send the `cache` in.